transports/webtransport/src/certificate.rs

```diff
@@ -0,0 +1,167 @@
+use std::{
+    io,
+    io::{Cursor, Read, Write},
+};
+
+use libp2p_core::multihash::Multihash;
+use libp2p_tls::certificate;
+use rustls::pki_types::{CertificateDer, PrivateKeyDer};
+use sha2::Digest;
+use time::{Duration, OffsetDateTime};
+
+const MULTIHASH_SHA256_CODE: u64 = 0x12;
+const CERT_VALID_PERIOD: Duration = Duration::days(14);
+
+pub type CertHash = Multihash<64>;
+
+// I would like to avoid interacting with the file system as much as possible.
+// My suggestion would be:
+// - libp2p::webtransport::Transport::new takes a list of certificates (of type
+//   libp2p::webtransport::Certificate)
+// - libp2p::webtransport::Certificate::generate allows users generate a new certificate with
+//   certain parameters (validity date etc)
+// - libp2p::webtransport::Certificate::{parse,to_bytes} allow users to serialize and deserialize
+//   certificates
+#[derive(Debug, PartialEq, Eq)]
+pub struct Certificate {
+    pub der: CertificateDer<'static>,
+    pub private_key_der: PrivateKeyDer<'static>,
+    pub not_before: OffsetDateTime,
+    pub not_after: OffsetDateTime,
+}
```

Comment:
Im not against the fields being public, but I do wonder if we should make them private and have getters and setters so if there are any additional fields added, it would not be require a minor version bump. We could probably even use `non_exhaustive` attribute. Thoughts?

Reply:
Ok. I think, there is no needing for setters at this time.